src/cpp/utility.cpp

```diff
@@ -27,30 +27,29 @@
  */
 
 #include <cstdlib>
+#include <cctype>
+#include <algorithm>
 #include <string>
 #include <cstring>
 #include <stdio.h>
 #include <stdexcept>
+#include <iostream>
+#include <fstream>
 #include "srexception.h"
 #include "utility.h"
 #include "logger.h"
 
 namespace SmartRedis {
 
-/*!
-*   \brief Initialize an integer from configuration, such as an
-*          environment variable
-*   \param value Receives the configuration value
-*   \param cfg_key The key to query for the configuration variable
-*   \param default_value Default if configuration key is not set
-*   \param suppress_warning Do not issue a warning if the variable
-*                           is not set
-*/
+// Initialize an integer from an environment variable
 void get_config_integer(int& value,
                         const std::string& cfg_key,
                         const int default_value,
-                        bool suppress_warning /*= false*/)
+                        int flags /* = 0 */)
 {
+    bool suppress_warning = 0 != (flags & flag_suppress_warning);
+    bool no_default = 0 != (flags & flag_no_default);
```

Comment:
Parsing too many nots is a bit challening. Can you change the name to `has_default` and reverse the logic?

Reply:
Renamed to throw_on_absent to explain the meaning better and eliminate double negative